Cargo.toml

```diff
@@ -32,15 +32,23 @@ row-serialize = ["serde"]
 hecs-macros = { path = "macros", version = "0.9.1", optional = true }
 hashbrown = { version = "0.14", default-features = false, features = ["ahash", "inline-more"] }
 serde = { version = "1.0.117", default-features = false, optional = true }
-spin = { version = "0.9.8", default-features = false, features = ["mutex", "spin_mutex"] }
+spin = { version = "0.9.8", default-features = false, features = ["mutex", "spin_mutex", "lazy"] }
 
-[dev-dependencies]
+[target.'cfg(not(target_os = "none"))'.dev-dependencies]
 bencher = "0.1.5"
 rand = "0.8.3"
 trybuild = "1.0.23"
 serde = { version = "1.0.117", features = ["derive"] }
 serde_test = "1.0.117"
 
+[target.'cfg(target_os = "none")'.dev-dependencies]
+
+# An example to demonstrate that the macros crate can be used without std.
+# This example is mainly used for testing.
+[[example]]
+name = "no_std_macros"
+crate-type = ["staticlib"]
```

Comment:
I think you need to add `required-features = ["macros"]` to avoid build failures in `no-default-features` builds.

Reply:
This example is removed. It's now part of the integration test.